tensorflow_probability/python/distributions/sigmoid_beta_test.py

```diff
@@ -0,0 +1,244 @@
+    a = tf.constant(1.0)
+    b = tf.constant(2.0)
+    dist = tfd.SigmoidBeta(a, b, validate_args=True)
+
+    self.assertAllClose(self.evaluate(tf.math.log(a / b)),
+                        self.evaluate(dist.mode()))
+
+  def testAssertsPositiveConcentration1(self):
+    concentration1 = tf.Variable([1., 2., -3.])
+    self.evaluate(concentration1.initializer)
+    with self.assertRaisesOpError('Concentration parameter must be positive.'):
+      d = tfd.SigmoidBeta(concentration1=concentration1,
+                          concentration0=[5.],
+                          validate_args=True)
+      self.evaluate(d.sample(seed=test_util.test_seed()))
+
+  def testAssertsPositiveConcentration1AfterMutation(self):
+    concentration1 = tf.Variable([1., 2., 3.])
+    self.evaluate(concentration1.initializer)
+    d = tfd.SigmoidBeta(concentration1=concentration1,
+                        concentration0=[5.],
+                        validate_args=True)
+    with self.assertRaisesOpError('Concentration parameter must be positive.'):
+      with tf.control_dependencies([concentration1.assign([1., 2., -3.])]):
+        self.evaluate(d.sample(seed=test_util.test_seed()))
+
+  @test_util.tf_tape_safety_test
+  def testGradientThroughConcentration0(self):
+    concentration0 = tf.Variable(3.)
+    d = tfd.SigmoidBeta(concentration0=concentration0,
+                        concentration1=5.,
+                        validate_args=True)
+    with tf.GradientTape() as tape:
+      loss = -d.log_prob([0.25, 0.5, 0.9])
+    grad = tape.gradient(loss, d.trainable_variables)
+    self.assertLen(grad, 1)
+    self.assertAllNotNone(grad)
+
+  def testAssertsPositiveConcentration0(self):
+    concentration0 = tf.Variable([1., 2., -3.])
+    self.evaluate(concentration0.initializer)
+    with self.assertRaisesOpError('Concentration parameter must be positive.'):
+      d = tfd.SigmoidBeta(concentration0=concentration0,
+                          concentration1=[5.],
+                          validate_args=True)
+      self.evaluate(d.sample(seed=test_util.test_seed()))
+
+  def testAssertsPositiveConcentration0AfterMutation(self):
+    concentration0 = tf.Variable([1., 2., 3.])
+    self.evaluate(concentration0.initializer)
+    d = tfd.SigmoidBeta(concentration0=concentration0,
+                        concentration1=[5.],
+                        validate_args=True)
+    with self.assertRaisesOpError('Concentration parameter must be positive.'):
+      with tf.control_dependencies([concentration0.assign([1., 2., -3.])]):
+        self.evaluate(d.sample(seed=test_util.test_seed()))
+
+  def _kstest(self, a, b, samples):
+    # Uses the Kolmogorov-Smirnov test for goodness of fit.
+    oracle_dist = tfd.SigmoidBeta(concentration0=a, concentration1=b)
+    ks, _ = sp_stats.kstest(samples,
+                            lambda x: self.evaluate(oracle_dist.cdf(x)))
+    # Return True when the test passes.
+    return ks < 0.02
+
+  def testSample(self):
```

Comment:
Instead of doing a KS test, I might recommend using our statistical_testing library:
https://cs.opensource.google/tensorflow/probability/+/master:tensorflow_probability/python/distributions/half_student_t_test.py;drc=4fbb0a2c1bcd7c0de2ef77fbee5fd12257188833;l=603